### ISSUE TITLE
OpenAPI spec not generated with @Tag, inheritance and multiple http verbs

### ISSUE THREAD

**Issue by guskuma** [smallrye/smallrye-open-api#297]
I'm trying to migrate from WF18/Swagger to WF19/Microprofile-OpenAPI. Changing annotations is not a big deal but I found myself lost with no OpenAPI spec showing off. After some debugging I realized I could only see any spec when commenting Tag annotation, removing inheritance (Schema using oneOf property and using only POST http verb in my endpoint resource class.

When debugging (before commenting, removing inheritance and using only POST) I reached writeSchemaList method on smallrye OpenApiSerializer with models variable with size equals 2 but all elements null. I believe it's a bug.

My abstract class:
```
@Table(schema = "main")
@Entity
@Inheritance(strategy = InheritanceType.SINGLE_TABLE)
@DiscriminatorColumn(name = "tipo", discriminatorType = DiscriminatorType.STRING, length = 50)
@NamedQueries({
        @NamedQuery(name = Pagamento.QUERY_GET_BY_CLIENTID, query = "SELECT p FROM Pagamento p LEFT JOIN p.movimentacoes WHERE p.client = :client AND p.clientId = :clientId"),
        @NamedQuery(name = Pagamento.QUERY_GET_BY_ID, query = "SELECT p FROM Pagamento p LEFT JOIN p.movimentacoes WHERE p.client = :client AND p.id = :pagamentoId")
})
@XmlRootElement
@XmlAccessorType(XmlAccessType.NONE)
@JsonTypeInfo(
        use = JsonTypeInfo.Id.NAME,
        include = JsonTypeInfo.As.EXISTING_PROPERTY,
        property = "tipo",
        visible = true)
@JsonSubTypes({
        @JsonSubTypes.Type(value = PagamentoTransferencia.class, name = TipoPagamento.Values.TRANSFERENCIA),
        @JsonSubTypes.Type(value = PagamentoTitulo.class, name = TipoPagamento.Values.TITULO)
})
public abstract class Pagamento implements Identifiable<Long> {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Schema(example = "2432", description = "Id do pagamento na API de Pagamentos", readOnly = true)
    @XmlAttribute
    private Long id = null;

// lot of properties...
}
```

My subclass 1
```
@Entity
@DiscriminatorValue(TipoPagamento.Values.TRANSFERENCIA)
@XmlRootElement
@XmlAccessorType(XmlAccessType.NONE)
public class PagamentoTransferencia extends Pagamento {
// more properties
}
```

My subclass 2
```
@Entity
@DiscriminatorValue(TipoPagamento.Values.TITULO)
@XmlRootElement
@XmlAccessorType(XmlAccessType.NONE)
public class PagamentoTitulo extends Pagamento {
// even more properties
}
```

My endpoint (only POST but using inheritance it doesn't matter if i have one or more http verbs)
```
@Tag(name = "Pagamentos-v1", description = "Operações sobre a entidade Pagamento")
@Path("/v1/pagamentos")
@ApplicationScoped
public class PagamentoResource implements Serializable {

    private static final long serialVersionUID = -8952762584675596958L;

    @Inject
    private PagamentoService pagamentoService;

    @Inject
    private MovimentacaoPagamentoService movimentacaoPagamentoService;

    @Inject
    BancoService bancoService;


    @Context
    ServletContext servletContext;

    @POST
    @Consumes({MediaType.APPLICATION_JSON})
    @Produces({MediaType.APPLICATION_JSON})
    @Operation(summary = "Cria novo pagamento", description = "Cria novo pagamento",
            operationId = "criarPagamento")
    @APIResponses(value = {
            @APIResponse(responseCode = "201", description = "Created",
                    content = @Content(schema = @Schema(implementation = Pagamento.class, oneOf = {PagamentoTransferencia.class, PagamentoTitulo.class}, discriminatorProperty = "tipo")),
                    links = {@org.eclipse.microprofile.openapi.annotations.links.Link(ref = "self", operationId = "retornarPagamento",
                                    parameters = {@LinkParameter(name = "pagamentoId", expression = "$response.body#id")}),
                            @org.eclipse.microprofile.openapi.annotations.links.Link(ref = "movimentações", operationId = "retornarMovimentacoes",
                                    parameters = {@LinkParameter(name = "pagamentoId", expression = "$response.body#id")})}),
            @APIResponse(responseCode = "400", description = "Bad Request", content = @Content(schema = @Schema(implementation = DefaultRestException.class))),
            @APIResponse(responseCode = "403", description = "Forbidden", content = @Content(schema = @Schema(implementation = DefaultRestException.class)))
    })
    public Response criarPagamento(@Parameter(schema = @Schema(implementation = Pagamento.class, oneOf = {PagamentoTransferencia.class, PagamentoTitulo.class}, discriminatorProperty = "tipo"), description = "Informações do pagamento a ser incluído. ATENÇÃO: Existe uma chave UNIQUE para client e clientId.", required = true) Pagamento pagto,
                                   @Context SecurityContext securityContext) {
      // code
    }
}
```

I tryed to specity discriminatorMapping on Schema annotation since class Name does not match my discriminatorProperty value but with no luck, same error: models variable with size equals 2 and null values.

Hopefully my case can help smallrye OpenAPI evolve.

Thanks

**Comment by MikeEdgar** [comment]
@guskuma - I did a small test using the classes you provided, but it seems to be generating the OpenAPI output as expected. Can you provide a sample project that demonstrates the problem? Thanks

**Comment by guskuma** [comment]
Hi @MikeEdgar I'll remove some private dependencies from my project to submit to you. Please just wait a bit.

**Comment by guskuma** [comment]
@MikeEdgar I removed some private dependencies and leaved others that might be interfering in OpenAPI spec generation.

[pagamentos.zip](https://github.com/smallrye/smallrye-open-api/files/4531491/pagamentos.zip)

**Comment by MikeEdgar** [comment]
Thanks @guskuma. Is the expectation with the project you attached that no OpenAPI is generated or that it is somehow incorrect?

**Comment by guskuma** [comment]
@MikeEdgar no spec is generated. All I have is a blank document.

**Comment by MikeEdgar** [comment]
@guskuma - please try updating to Wildfly 19.1. I think you ran into this issue which has been fixed with the latest release.

https://issues.redhat.com/browse/WFLY-13333

**Comment by guskuma** [comment]
OK @MikeEdgar. I'll try and let you know as soon as I can.

**Comment by MikeEdgar** [comment]
@guskuma - were you able to update Wildfly? Please let us know if this is still an issue for you with the current version.

**Comment by MikeEdgar** [comment]
@guskuma - I am going to go ahead and close this issue as it looks like it was a Wildfly bug that was fixed in 19.1 (https://issues.redhat.com/browse/WFLY-13333). If you disagree or still have the problem after upgrading Wildfly, please feel free to re-open this and we can continue to research.
